src/session/sessions.ts

```diff
@@ -0,0 +1,284 @@
+   *
+   * @param {string | undefined} value A users JWT
+   * @description Saves the current token to cookies. If the value is undefined,
+   *              the cookie will be removed. The expire time is set to four
+   *              minutes.
+   */
+  set token(value: string | undefined) {
+    if (value) {
+      Cookies.set(this.tokenName, value ? value : "", {
+        /* Expire time is set to 4 minutes */
+        expires: 4 / 1440,
+      });
+    } else {
+      Cookies.remove(this.tokenName);
+    }
+  }
+
+  /**
+   * Write refresh token to cookies.
+   *
+   * @param {string | undefined} value A users JWT refresh token
+   * @description Saves the current refresh token to cookies. If the value
+   *              is undefined, the cookie will be removed. The expire time is
+   *              set to six days.
+   */
+  set refreshToken(value: string | undefined) {
+    if (value) {
+      Cookies.set(this.refreshTokenName, value, {
+        /* Expire time is set to 6 days */
+        expires: 6,
+      });
+    } else {
+      Cookies.remove(this.refreshTokenName);
+    }
+  }
+
+  //> Methods
+  /**
+   * Get a valid session token. If there is not the session will be refreshed.
+   *
+   * @returns {Promise<string | undefined>} The session token if set
+   */
+  async upToDateToken(): Promise<string | undefined> {
+    let token = super.token;
+
+    /* Refresh token if there is none */
+    if (!token) {
+      await this.refresh();
+
+      token = super.token;
+    }
+
+    return token;
+  }
+
+  /**
+   * Send a query to the endpoint.
+   *
+   * @param {string} token A authentication token
+   * @param {DocumentNode} data A DocumentNode with a query
+   * @param {object} variables A object with variables
+   */
+  async send(token: string, data: DocumentNode, variables?: object) {
+    let headers = {
+      authorization: token,
+    };
+
+    return this.ep.sendQuery(data, variables, headers);
+  }
+
+  /**
+   * Begin session.
+   *
+   * @param {string} user A User defined by username and password
+   * @returns {Promise<any>} A UserData object
+   */
+  async begin(user?: User): Promise<any> {
+    let response;
+
+    if (!user && this.refreshToken) {
+      /* Refresh token and retrieve a new refreshToken if necessary */
+      this.refresh();
+    } else {
+      if (!user) {
+        /* Authenticate anonymous user */
+        response = await this.tasks.auth.anon();
+      } else {
+        /* Authenticate real user */
+        response = await this.tasks.auth.nonanon(user);
+      }
+
+      /* Set tokens */
+      this.token = response.data?.auth.token;
+      this.refreshToken = response.data?.auth.refreshToken;
+
+      return response.data?.auth.user;
+    }
+
+    /* Get user data */
+    response = await this.tasks.user.whoami();
+
+    return response.data?.whoami;
+  }
+
+  /**
+   * Refreshes a session based on its history.
+   *
+   * @description When there is no token the refresh task is called. When there
```

Comment:
Change to:"
If there is no token the refresh token is used to get a new one. 
If none of both is provided begin a new session as an anonymous user."